semantic-conventions/src/opentelemetry/semconv/model/semantic_convention.py

```diff
@@ -247,7 +247,48 @@ def __init__(self, group):
 class MetricSemanticConvention(BaseSemanticConvention):
     GROUP_TYPE_NAME = "metric"
 
-    allowed_keys = ()
+    allowed_keys: Tuple[str, ...] = BaseSemanticConvention.allowed_keys + (
+        "metric_name",
+        "units",
+        "instrument",
+    )
+
+    yaml_to_markdown_instrument_repr: Dict[str, str] = {
+        "counter": "Counter",
+        "updowncounter": "UpDownCounter",
+        "histogram": "Histogram",
+        "gauge": "Gauge",
+    }
+
+    allowed_instruments: Tuple[str, ...] = tuple(
+        yaml_to_markdown_instrument_repr.keys()
+    ) + (None,)
+
+    def __init__(self, group):
+        super().__init__(group)
+        self.name = group.get("metric_name")
+        self.units = group.get("units")
+        self.instrument = group.get("instrument")
+        self.instrument_markdown_fmt = self.yaml_to_markdown_instrument_repr.get(
+            self.instrument
+        )
+        self.validate()
+
+    def validate(self):
+        val_tuple = (self.name, self.units, self.instrument)
+        if not all(val_tuple) and any(val_tuple):
+            raise ValidationError.from_yaml_pos(
+                self._position,
+                "Either all or none of name, units, and instrument must be defined",
```

Comment:
The case where none of them are defined are the "parent" metrics in the yaml which aren't actually metrics. For example, in the test `metric.foo` is one of these. 

These exist for two reasons: 
- So multiple metrics can have a common parent and follow the same inheritance pattern that spans follow. 
- So attributes that aren't defined elsewhere can be defined on the parent without tying them to a specific metric.

Reply:
To clarify a bit more, this is the same as what happens in the span semantic convention currently. For example, the [network group](https://github.com/open-telemetry/opentelemetry-specification/blob/main/semantic_conventions/trace/general.yaml#L2) exists so it can group attributes related to networks.

Reply:
Ah makes sense, got it. Thanks for the explanation, you can resolve this conversation :)